contracts/ArrowToken.sol

```diff
@@ -8,17 +8,26 @@ import "@openzeppelin/contracts-upgradeable/proxy/utils/UUPSUpgradeable.sol";
 
 /**
     @title Arrow Token Implementation
+    @notice This is an upgradeable contract that conforms to the ERC-20 standard
+    @dev Upgrades should be made in a separate implementation contract; see "./test/ArrowTokenV2.sol"
```

Comment:
This is an interesting question that I haven't thought through. Do upgrades need to be made in a separate file? We could just update the existing file and upgrade that way.

Reply:
Based on most of the online tutorials I've seen, they usually create a separate file with different naming system. I think maybe it's easier to and deploy, and potentially avoid any string collisions?

Reply:
Not sure what you mean by string collisions?.

I think there's a difference between online tutorials and production systems. Can you cite any examples from live projects that follow this convention? 

I would lean towards upgrading the interface in the same file, since you get versioning history built in from the git commits that are easier to understand.

Reply:
Ok that makes sense. I couldn't find any live projects that implement UUPS proxies. We will upgrade directly from the source file then, and see how it goes as the project grows.